jenkins/check-for-build.jenkinsfile

```diff
@@ -40,6 +40,11 @@ pipeline {
             description: 'Job to trigger if build has changed',
             trim: true
         )
+        string(
```

Comment:
@rishabh6788 the reason we choose to use string in both OS and OSD build is because choice param requires me to add `linux, macos, windows, linux+macos, macos+windows, linux+windows, linux+macos+windows` entries.

If we later on add a new distribution this would be even more complicated.

No worries as the verify section already verifying the entry and will fail the build if there is any human errors.
Thanks.